source/docs/magento1/add-custom-endpoint.md

```diff
@@ -0,0 +1,251 @@
+
+### Example
+
+```
+<?php
+
+/**
+ * Class Module_Network_Model_Api2_Post_Rest_Guest_V1
+ */
+class Module_Network_Model_Api2_Post_Rest_Guest_V1 extends FrontCommerce_Integration_Model_Api2_Abstract
+{
+    /**
+     * Get post list
+     *
+     * @return array
+     */
+    protected function _retrieveCollection()
+    {
+        $collection = $this->_getCollection();
+        $this->_applyCollectionModifiers($collection);
+        $this->_loadCollection($collection);
+        $this->addCacheHeaders($collection->getCacheLifetime());
+        $data = $collection->walk('toArray');
+        return array_values((array) $data);
+    }
+
+    /**
+     * Retrieve post collection
+     *
+     */
+    protected function _getCollection()
+    {
+        $collection = Mage::getResourceModel('module_network/post_collection');
+        return $collection;
+    }
+
+    /**
+     * Retrieve information about specified socialize
+     *
+     * @throws Mage_Api2_Exception
+     * @return array
+     */
+    protected function _retrieve()
+    {
+        $entityId = this->getRequest()->getParam('id'); // param name is defined on your route node, for this example is :id
+        /* @var $stockItem Mage_CatalogInventory_Model_Stock_Item */
+        $post = Mage::getModel('module_network/post')->load($entityId);
+        if (!$post || !$post->getId() || $post->getId() != $entityId) {
+            $this->_critical(self::RESOURCE_NOT_FOUND);
+        }
+        return $post->getData();
+    }
+}
+
+```
+
+### Testing
+
+In case you need to test your endpoints, you can either use cURL or [Postman](https://www.getpostman.com/). But advanced clients like Postman will allow you to have credentials and will make it easier to test customers vs guest endpoints.
+
+- Guest testing : If your endpoint can be access in guest mode, you can simply send GET / POST / DELETE / UPDATE request
+  to your endpoint for see the response
+- Customer testing : You need to add credential and token for your request. You can retrieve all this informations in your database
+
+  - Consumer Key = `key` in table `oauth_consumer`
+  - Consumer Secret = `secret` in table `oauth_consumer`
+  - Access Token = `token` of user in table `oauth_token`
+  - Token secret = `secret` of same user in table `oauth_token`
+    ![The Authorization tab of a Postman request with OAuth 1.0 selected.](./assets/testing.png)
+
+## Good to know
+
+- If you can, extend `FrontCommerce_Integration_Model_Api2_Abstract` in your own API class. This class add useful functions like:
+    - `public function getCustomer()` : Retrieve current customer and save it in customer session.
+    - `protected function _initStore()` : Set current store with default store view or store set in request params.
+    - `protected function _getStore()` : Rewrite Magento get store base method for no repeat function if store was already set. Retrieve current store according to request and API user type.
```

Comment:
I am not sure my understanding is correct, but I suggest the following sentence (easier to read IMO):

```suggestion
    - `protected function _getStore()` : Rewrite Magento's base method to memoize store value. Retrieve current store according to request and API user type.
```